### PR TITLE
create gist and preview output on build failures

### PR DESCRIPTION
**Pull request by ewlu**
manual run with build-failure: 
~https://github.com/ewlu/gcc-precommit-ci/actions/runs/11389822472~
~https://github.com/ewlu/gcc-precommit-ci/actions/runs/11390176809~
~https://github.com/ewlu/gcc-precommit-ci/actions/runs/11390582485~
https://github.com/ewlu/gcc-precommit-ci/actions/runs/11390649510

### CODE REVIEW

File: .github/actions/report-build-complete-status/action.yaml [hunk 1]
```diff
@@ -50,13 +50,40 @@ runs:
           python scripts/update_issue_status.py -token ${{ inputs.github_token }} -state 'Success' -comment ${{ inputs.build_comment_id }} -target ${{ inputs.target }} -check 'Build GCC' -repo ewlu/gcc-precommit-ci -baseline $HASH
           printf "\n[Additional information](${{ inputs.additional_info }})\n" >> comment.md
 
+      - name: Build failure gist
+        if: ${{ inputs.build_status == 'failure' }}
+        shell: bash
+        working-directory: riscv-gnu-toolchain
+        run: |
+          cp ${{ inputs.build_log_name }}.log gist.md
```
Comment (kevinl-rivosinc) [inline .github/actions/report-build-complete-status/action.yaml:58]: Wouldn't it be faster to directly `tail -c $ONE_MB_IN_BYTES ${{ inputs.build_log_name }}.log > gist.md`? Tail would automatically add the entire file if it's less than 1mb and truncate if it's greater than 1mb.

File: .github/actions/report-build-complete-status/action.yaml [hunk 2]
```diff
@@ -50,13 +50,40 @@ runs:
...
+          cat gist.md
+
+      - name: Create build failure gist
+        if: ${{ inputs.build_status == 'failure' }}
+        id: gist-url
+        shell: bash
+        working-directory: riscv-gnu-toolchain
+        run: |
+          if [ -e "gist.md" ]; then
+            python3 ./scripts/create_gist.py --token ${{ secrets.GIST_TOKEN }} --input gist.md --output gist_url.txt --title ${{ inputs.issue_num }}_${{ inputs.patch_name }}_${{ inputs.build_log_name }}.md
```
Comment (kevinl-rivosinc) [inline .github/actions/report-build-complete-status/action.yaml:74]: Does secrets get inherited for the action?

File: .github/actions/report-build-complete-status/action.yaml [hunk 3]
```diff
@@ -50,13 +50,40 @@ runs:
...
+        continue-on-error: true
+
       - name: Build failure report
         if: ${{ inputs.build_status == 'failure' }}
         shell: bash
         working-directory: riscv-gnu-toolchain
         run: |
           if [ '${{ inputs.build_only }}' == 'true' ]; then export HASH=${{ inputs.tot_hash }}; else export HASH=${{ inputs.baseline_hash }}; fi
-          python scripts/update_issue_status.py -token ${{ inputs.github_token }} -state 'Build failure. Please check the ${{ inputs.build_log_name }} artifact' -comment ${{ inputs.build_comment_id }} -target ${{ inputs.target }} -check 'Build GCC' -repo ewlu/gcc-precommit-ci -baseline $HASH
+          python scripts/update_issue_status.py -token ${{ inputs.github_token }} -state 'Build failure. Please check the ${{ inputs.build_log_name }} artifact. [Preview](${{ steps.gist-url.outputs.gist_url }})' -comment ${{ inputs.build_comment_id }} -target ${{ inputs.target }} -check 'Build GCC' -repo ewlu/gcc-precommit-ci -baseline $HASH
```
Comment (patrick-rivos) [inline .github/actions/report-build-complete-status/action.yaml:86]: Nit on the link verbage:
```suggestion
          python scripts/update_issue_status.py -token ${{ inputs.github_token }} -state 'Build failure. Please check the ${{ inputs.build_log_name }} artifact. [Tail Preview (1MB)](${{ steps.gist-url.outputs.gist_url }})' -comment ${{ inputs.build_comment_id }} -target ${{ inputs.target }} -check 'Build GCC' -repo ewlu/gcc-precommit-ci -baseline $HASH
```

File: .github/actions/report-build-complete-status/action.yaml (hunk 1 above)
Comment (ewlu) [reply .github/actions/report-build-complete-status/action.yaml]: yeah that's true. updated

File: .github/actions/report-build-complete-status/action.yaml (hunk 2 above)
Comment (ewlu) [reply .github/actions/report-build-complete-status/action.yaml]: they shouldn't. forgot about this

File: .github/actions/report-build-complete-status/action.yaml [hunk 4]
```diff
@@ -50,13 +53,35 @@ runs:
...
+          tail -c $ONE_MB_IN_BYTES ${{ inputs.build_log_name }}.log > gist.md
+          cat gist.md
+
+      - name: Create build failure gist
+        if: ${{ inputs.build_status == 'failure' }}
+        id: gist-url
+        shell: bash
+        working-directory: riscv-gnu-toolchain
+        run: |
+          if [ -e "gist.md" ]; then
```
Comment (patrick-rivos) [inline .github/actions/report-build-complete-status/action.yaml:71]: Do we need this?

Comment (patrick-rivos) [reply .github/actions/report-build-complete-status/action.yaml:71]: And can we fold this into the above block?

**Review by patrick-rivos — APPROVED** [review]

**Review by kevinl-rivosinc — APPROVED** [review]

File: .github/actions/report-build-complete-status/action.yaml [hunk 5]
```diff
@@ -50,13 +53,29 @@ runs:
...
+        if: ${{ inputs.build_status == 'failure' }}
+        id: gist-url
+        shell: bash
+        working-directory: riscv-gnu-toolchain
+        run: |
+          ONE_MB_IN_BYTES=1048000
+          echo "\`\`\`" > gist.md
+          tail -c $ONE_MB_IN_BYTES build/${{ inputs.build_log_name }}.log >> gist.md
+          python3 ./scripts/create_gist.py --token ${{ inputs.gist_token }} --input gist.md --output gist_url.txt --title ${{ inputs.issue_num }}_${{ inputs.patch_name }}_${{ inputs.build_log_name }}.md
+          echo "\`\`\`" >> gist.md
```
Comment (patrick-rivos) [inline .github/actions/report-build-complete-status/action.yaml:66]: This needs to go above the script in order to be included in the generated gist

**Comment by ewlu** [comment]
Issue: https://github.com/ewlu/gcc-precommit-ci/issues/2427

**Review by patrick-rivos — APPROVED** [review]

**Pull request merged**